plugins/eni/e2eTests/e2e_test.go

```diff
@@ -0,0 +1,334 @@
+	ec2Client := ec2.New(session.Must(session.NewSession()), &aws.Config{
+		Region: aws.String(cfg.region),
+	})
+
+	// Create an ENI
+	eni, err := createENI(ec2Client, cfg)
+	require.NoError(t, err, "Unable to create ENI")
+	defer deleteENI(ec2Client, eni)
+
+	require.NoError(t, waitUntilNetworkInterfaceAvailable(ec2Client, eni), "ENI didn't transition into 'available'")
+	// Attach the ENI to the instance
+	attachment, err := attachENI(ec2Client, cfg, eni)
+	require.NoError(t, err, "Unable to attach ENI")
+	defer detachENI(ec2Client, attachment)
+
+	waitUntilNetworkInterfaceAttached(eni, 5*time.Second)
+
+	// Create a directory for storing test logs
+	testLogDir, err := ioutil.TempDir("", "ecs-eni-e2e-test-")
+	require.NoError(t, err, "Unable to create directory for storing test logs")
+
+	// Configure the env var to use the test logs directory
+	os.Setenv("ECS_CNI_LOG_FILE", fmt.Sprintf("%s/eni.log", testLogDir))
+	t.Logf("Using %s for test logs", testLogDir)
+	defer os.Unsetenv("ECS_CNI_LOG_FILE")
+
+	// Configure dhclient env var's for the plugin to use test logs directory
+	os.Setenv("ENI_DHCLIENT_LEASES_PATH", testLogDir)
+	defer os.Unsetenv("ENI_DHCLIENT_LEASES_PATH")
+	os.Setenv("ENI_DHCLIENT_PID_FILE_PATH", testLogDir)
+	defer os.Unsetenv("ENI_DHCLIENT_PID_FILE_PATH")
+
+	// Handle deletion of test logs at the end of the test execution if
+	// specified
+	ok, err := strconv.ParseBool(getEnvOrDefault("ECS_PRESERVE_E2E_TEST_LOGS", "false"))
+	assert.NoError(t, err, "Unable to parse ECS_PRESERVE_E2E_TEST_LOGS env var")
+	if !ok {
+		defer os.RemoveAll(testLogDir)
+	}
+
+	// Use the current network namespace to execute the test in
+	testNS, err := ns.GetCurrentNS()
+	require.NoError(t, err, "Unable to get the network namespace to run the test in")
+	defer testNS.Close()
+
+	// Create a network namespace to mimic the container's network namespace.
+	// The ENI will be moved to this namespace
+	targetNS, err := ns.NewNS()
+	require.NoError(t, err,
+		"Unable to create the network namespace that represents the network namespace of the container")
+	defer targetNS.Close()
+
+	// Construct args to invoke the CNI plugin with
+	execInvokeArgs := &invoke.Args{
+		ContainerID: containerID,
+		NetNS:       targetNS.Path(),
+		IfName:      ifName,
+		Path:        os.Getenv("CNI_PATH"),
+	}
+	netConf := []byte(fmt.Sprintf(netConfFormat,
+		aws.StringValue(eni.NetworkInterfaceId),
+		aws.StringValue(eni.PrivateIpAddress),
+		aws.StringValue(eni.MacAddress)))
+	t.Logf("Using config: %s", string(netConf))
+
+	testNS.Do(func(ns.NetNS) error {
+		// Execute the "ADD" command for the plugin
+		execInvokeArgs.Command = "ADD"
+		err := invoke.ExecPluginWithoutResult(
+			eniPluginPath,
+			netConf,
+			execInvokeArgs)
+		require.NoError(t, err, "Unable to execute ADD command for ecs-eni plugin")
+		return nil
+	})
+
+	targetNS.Do(func(ns.NetNS) error {
+		// Validate that only 2 devices exist in the target network
```

Comment:
Yes, but there's no sane way of doing that. We can read the pid value from the file as stop-gap, but even that's not super clean I feel. I'll add a TODO as placeholder until we figure out a way to do that.